File: packages/gbnf-experiment/data/20260909T023701Z_e94dca40/ported_implementation/src/utils/errors/get-input-as-string.ts
import type { ValidInput } from './errors-types.js';

export const getInputAsString = (src: ValidInput): string => {
  if (typeof src === 'string') {
    return src;
  }
  if (typeof src === 'number') {
    return String.fromCodePoint(src);
  }
  return src.map((cp) => String.fromCodePoint(cp)).join('');
};

File: packages/gbnf-experiment/data/20260908T234247Z_2da5a2c0/ported_implementation/gbnf/utils/errors/get_input_as_string.py
"""Port of ``src/utils/errors/get-input-as-string.ts``."""

from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:  # pragma: no cover
    from ...grammar_graph.types import ValidInput


def get_input_as_string(src: "ValidInput") -> str:
    if isinstance(src, str):
        return src
    code_points = list(src) if isinstance(src, (list, tuple)) else [src]
    return "".join(chr(cp) for cp in code_points)
